src/FileAbstraction/ReplacingFile.php

```diff
@@ -5,16 +5,54 @@
 namespace Vich\UploaderBundle\FileAbstraction;
 
 use Symfony\Component\HttpFoundation\File\File;
-use Symfony\Component\HttpFoundation\File\UploadedFile;
 
 /**
  * This class can be used to signal that the given file should be "uploaded" into the Vich-abstraction
  * in cases where it is not possible to construct an `UploadedFile`.
  */
 class ReplacingFile extends File
 {
+    protected bool $removeReplacedFile;
+    protected bool $removeReplacedFileOnError;
+
+    public function __construct(
+        string $path,
+        bool $checkPath = true,
+        bool $removeReplacedFile = false,
+        bool $removeReplacedFileOnError = false
+    ) {
+        parent::__construct($path, $checkPath);
+
+        $this->removeReplacedFile = $removeReplacedFile;
+        $this->removeReplacedFileOnError = $removeReplacedFileOnError;
+    }
+
     public function getClientOriginalName(): string
     {
         return $this->getFilename();
     }
+
+    public function isRemoveReplacedFile(): bool
+    {
+        return $this->removeReplacedFile;
+    }
+
+    public function setRemoveReplacedFile(bool $removeReplacedFile): self
```

Comment:
I don't see any use for this setter

Reply:
You can use parameter passing in the constructor, or you can call this setter.
```
new ReplacingFile($file, true, true, true)
```
or
```
$replacingFile = new ReplacingFile($file);
$replacingFile->setRemoveReplacedFile(true);
```
or
```
(new ReplacingFile($file))->setRemoveReplacedFile(true);
```
Whichever is more convenient

Reply:
Still, I don't see a case where calling the setter could be convenient. If the only purpose is offering a setter alternative to the constructor, it's a weak purpose. It's better to keep the object immutable.

Reply:
Removed the setters